src/bridge/rnastructure.cpp

```diff
@@ -38,8 +39,45 @@ std::vector<Secondary> StructureToSecondarys(structure& struc) {
   return s;
 }
 
+std::vector<subopt::SuboptResult> StructureToSuboptVector(structure& struc) {
+  auto s_list = StructureToSecondarys(struc);
+  std::vector<subopt::SuboptResult> res;
+  for (int i = 0; i < static_cast<int>(s_list.size()); ++i) {
+    // TODO: Convert CTDs?
+    res.emplace_back(subopt::SuboptResult(
+        Energy(struc.GetEnergy(i + 1)), tb::TracebackResult(std::move(s_list[i]), Ctds())));
+  }
+  return res;
+}
+
 }  // namespace
 
+RNAstructure::PartitionFunctionState::PartitionFunctionState(int N, datatable* data) {
```

Comment:
Can you initialize these via constructor initializer list?

Reply:
Done.